src/App.tsx

```diff
@@ -15,47 +16,124 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App: React.FC = () => {
+enum SortType {
+  NONE,
+  ALPHABET,
+  LENGTH,
+}
+
+type ReorderOptions = {
+  sortType: SortType,
+  isReversed: boolean,
+};
+
+export function getReorderedGoods(
+  goods: string[],
+  { sortType, isReversed }: ReorderOptions,
+) {
+  const visibleGoods = [...goods];
+
+  visibleGoods.sort((a, b) => {
+    switch (sortType) {
+      case SortType.ALPHABET:
+        return a.localeCompare(b);
+      case SortType.LENGTH:
+        return a.length - b.length;
+      default:
+        return SortType.NONE;
+    }
+  });
```

Comment:
if I add sort in every case, I also need to add reverse there - which will increase the amount of code and confuse logic - otherwise it doesn't work properly 

`switch (sortType) {
    case SortType.ALPHABET:
      return visibleGoods.sort((a, b) => a.localeCompare(b)) ;
    case SortType.LENGTH:
      return visibleGoods.sort((a, b) => a.length - b.length);
    default:
      return true;
  }

  if (isReversed) {
    visibleGoods.reverse();
  }

  return visibleGoods;`